pydm/main_window.py

```diff
@@ -498,3 +500,68 @@ def _confirm_quit(self, callback):
     def show_macro_window(self):
         macro_window = MacroWindow(self)
         macro_window.show()
+
+    def add_menu_items(self):
+        # create the custom menu with user given items
+        if not isinstance(self.display_widget(), Display):
+            return
+        items = self.display_widget().menu_items()
+        if len(items) != 0:
+            self.create_menu(self.ui.menuCustomActions, items)
```

Comment:
I'd probably add an else case here with something like 
`self.ui.menuCustomActions.menuAction().setVisible(False)`
Just to prevent showing an Actions menu that when the user clicks on it doesn't expand to anything because no actions were actually added to it